src/mmw/js/src/draw/views.js

```diff
@@ -136,6 +136,85 @@ var DrawWindow = Marionette.ItemView.extend({
 
     id: 'draw-window',
 
+    regions: {
+        uploadRegion: '#file-upload-region'
+    },
+
+    ui: {
+        'selectBoundary': '#select-boundary-button',
+        'drawArea': '#draw-area-button',
+        'delineateWatershed': '#delineate-watershed-button',
+        'uploadFile': '#file-upload-button'
+    },
+
+    events: {
+        'click @ui.selectBoundary': 'toggleSelectBoundary',
+        'click @ui.drawArea': 'toggleDrawArea',
+        'click @ui.delineateWatershed': 'toggleDelineateWatershed',
+        'click @ui.uploadFile': 'toggleUploadFile'
+    },
+
+    modelEvents: {
+        'change': 'onShow'
+    },
+
+    templateHelpers: function() {
+        return {
+            activeDrawTool: this.model.get('selectedDrawTool'),
+        };
+    },
+
+    initialize: function() {
+        this.model.set('selectedDrawTool', null);
+    },
+
+    onShow: function() {
+        this.render();
```

Comment:
Consider dropping this function and having

```javascript
modelEvents: {
    'change': 'render'
},
```
`onShow` is called after `render` and is usually used to show child views.